### PR TITLE
Fix invalid references and unknown fields in twisted.internet

### PR DESCRIPTION
**Pull request by acabhishek942**
See: https://twistedmatrix.com/trac/ticket/8531

### CODE REVIEW

File: twisted/internet/cfreactor.py [hunk 1]
```diff
@@ -474,9 +478,9 @@ def install(runLoop=None, runner=None):
     @param runLoop: the run loop to use.
 
     @param runner: the function to call in order to actually invoke the main
-        loop.  This will default to L{CFRunLoopRun} if not specified.  However,
-        this is not an appropriate choice for GUI applications, as you need to
-        run NSApplicationMain (or something like it).  For example, to run the
+        loop.  This will default to L{CoreFoundation.CFRunLoopRun} if not specified.
+        However, this is not an appropriate choice for GUI applications, as you need
+        to run NSApplicationMain (or something like it).  For example, to run the
         Twisted mainloop in a PyObjC application, your C{main.py} should look
```
Comment (acabhishek942) [inline twisted/internet/cfreactor.py:484]: these invalid references can be fixed by adding links to CoreFoundation and CFNetwork

Comment (rodrigc) [inline twisted/internet/cfreactor.py:484]: OK.  @adiroiban submitted a similar fix to add zope links https://github.com/twisted/twisted/pull/238
Do you know what link we would need to add in a similar way for CoreFoundation and CFNetwork?

File: twisted/internet/defer.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ class CancelledError(Exception):
 class TimeoutError(Exception):
     """
     This exception is deprecated.  It is used only by the deprecated
```
Comment (rodrigc) [inline twisted/internet/defer.py:50]: Ah, you've encountered an interesting thing.  Part of the comment is wrong.  Just delete this sentence completely:

```
It is used only by the deprecated L{Deferred.setTimeout} method.
```

That comment is not true.  twisted.internet.defer.TimeoutError is used in other places, such as twisted/names/error.py
By working on epydoc problems you found some code that needs to be deprecated!!

I submitted this PR to do that: https://github.com/twisted/twisted/pull/261

File: twisted/internet/cfreactor.py (hunk 1 above)
Comment (adiroiban) [inline twisted/internet/cfreactor.py:484]: This can be fixed by adding an intersphinx link to  https://pythonhosted.org/pyobjc/objects.inv

Comment (rodrigc) [inline twisted/internet/cfreactor.py:484]: @adiroiban @acabhishek942  I found a utility [decode_sphinx_inventory.py](http://mbless.de/blog/2014/07/28/decode-sphinx-objects-inv/) which can analyze the contents of an intersphinx inventory file.   https://pythonhosted.org/pyobjc/objects.inv does **not** contain links to CoreFoundation, CFRunLoop, etc.

**Comment by rodrigc** [comment]
@acabhishek942 thanks for your assistance.  I was wondering if you could try some things out
and learn how to use code coverage.

Can you do the following?

1)  Make sure that tox and coverage Python packages are installed

```
pip install tox
pip install coverage
```

2)  Checkout the latest Twisted code from GitHub
3)  Go to the top level directory in the Twisted tree, and list all the tox environments

```
tox -l
```

These environments are defined in `tox.ini`

4)  Run the tests under tox, but with coverage

```
tox -e py27-coverage-posix
```

5)  After the tests have run, list all the coverage reports, by typing this in the top-level directory in the Twisted tree:

```
ls .coverage*
```

6)  Combine the coverage reports by typing:

```
coverage combine
```

You should now have a single `.coverage` file.

7)  Generate a plain coverage report by typing:

```
coverage report
```

8)  Generate a HTML based report by typing:

```
coverage html
```

Let me know if you can get this to work.

**Comment by acabhishek942** [comment]
@rodrigc I got this to work.

**Comment by rodrigc** [comment]
Nice!  So one thing Twisted devs are pushing for is that any PR to Twisted must have 100% code coverage of the patch itself.  This may require writing new tests.

**Pull request closed without merging**